Index: packagekit-0.5.4+20091104/etc/PackageKit.conf.in
===================================================================
--- packagekit-0.5.4+20091104.orig/etc/PackageKit.conf.in	2009-10-23 08:41:09.000000000 +0200
+++ packagekit-0.5.4+20091104/etc/PackageKit.conf.in	2009-11-05 10:40:27.000000000 +0100
@@ -115,14 +115,14 @@
 # NOTE: Don't enable this for backends that are slow doing SearchFile()
 #
 # default=true
-ScanDesktopFiles=true
+ScanDesktopFiles=false
 
 # Update the package list when we refresh the cache
 #
 # NOTE: Don't enable this for backends that are slow doing GetPackages()
 #
 # default=true
-UpdatePackageList=true
+UpdatePackageList=false
 
 # Check for running processes when we update packages
 #
@@ -130,7 +130,7 @@
 # installed files.
 #
 # default=true
-UpdateCheckProcesses=true
+UpdateCheckProcesses=false
 
 # Check for shared libraries that are in use, that are replaced by packages
 # that are marked as security updates.
--- packagekit/etc/PackageKit.conf.in	2010-01-05 09:17:53.000000000 +0000
+++ packagekit/etc/PackageKit.conf.in	2010-01-27 18:34:43.000000000 +0000
@@ -175,7 +175,7 @@
 # should be updated to reflect reality.
 #
 # default=30
-StateChangedTimeoutPriority=30
+StateChangedTimeoutPriority=60
 
 # The time in seconds to wait after the computer has been resumed or any non
 # package related system event
